File: web/src/api/proxyDownload.ts
import request from './request'

export interface ProxyDownloadLog {
  id: number
  repository_id: number
  repository?: {
    id: number
    name: string
    display_name: string
  }
  package_type: string
  package_name: string
  version: string
  filename: string
  remote_url: string
  status: string
  status_code: number
  size_bytes: number
  duration_ms: number
  from_cache: boolean
  ip_address: string
  user_agent: string
  user_id: number
  error_message: string
  created_at: string
}

export interface ProxyDownloadStats {
  total_downloads: number
  success_count: number
  failed_count: number
  cached_count: number
  total_bytes: number
}

export interface ProxyDownloadLogListResponse {
  items: ProxyDownloadLog[]
  pagination: {
    total: number
    page: number
    page_size: number
  }
}

export const proxyDownloadApi = {
  list(params?: { page?: number; page_size?: number; repository_id?: number; package_type?: string; status?: string; start_date?: string; end_date?: string }) {
    return request.get<ProxyDownloadLogListResponse>('/proxy-downloads/logs', { params })
  },

  getStats(params?: { repository_id?: number; start_date?: string; end_date?: string }) {
    return request.get<ProxyDownloadStats>('/proxy-downloads/stats', { params })
  },
}

File: internal/model/proxy_download.go
package model

import "time"

type ProxyDownloadLog struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	RepositoryID uint      `gorm:"not null;index:idx_log_repo_created_at,priority:1;index:idx_log_repo_status_created_at,priority:1" json:"repository_id"`
	Repository   *Repository `gorm:"foreignKey:RepositoryID" json:"repository,omitempty"`
	PackageType  string    `gorm:"size:20;not null;index" json:"package_type"`
	PackageName  string    `gorm:"size:200;not null;index" json:"package_name"`
	Version      string    `gorm:"size:50;index" json:"version,omitempty"`
	Filename     string    `gorm:"size:255" json:"filename,omitempty"`
	RemoteURL    string    `gorm:"size:500" json:"remote_url,omitempty"`
	Status       string    `gorm:"size:20;not null;index:idx_log_status;index:idx_log_repo_status_created_at,priority:2" json:"status"`
	StatusCode   int       `json:"status_code,omitempty"`
	SizeBytes    int64     `json:"size_bytes,omitempty"`
	DurationMs   int       `json:"duration_ms"`
	FromCache    bool      `gorm:"default:false;index" json:"from_cache"`
	IPAddress    string    `gorm:"size:45" json:"ip_address,omitempty"`
	UserAgent    string    `gorm:"size:500" json:"user_agent,omitempty"`
	UserID       *uint     `json:"user_id,omitempty"`
	ErrorMessage string    `gorm:"type:text" json:"error_message,omitempty"`
	CreatedAt    time.Time `gorm:"autoCreateTime;index:idx_log_created_at;index:idx_log_repo_created_at,priority:2;index:idx_log_repo_status_created_at,priority:3" json:"created_at"`
}

const (
	DownloadStatusSuccess = "success"
	DownloadStatusFailed  = "failed"
	DownloadStatusCached  = "cached"
)
